src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs

```diff
@@ -31,17 +32,25 @@ public partial class HttpConnectionManager
         private readonly ILogger<HttpConnectionManager> _logger;
         private readonly ILogger<HttpConnectionContext> _connectionLogger;
 
-        public HttpConnectionManager(ILoggerFactory loggerFactory, IApplicationLifetime appLifetime)
+        public TimeSpan DisconnectTimeout { get; set; }
+
+        public HttpConnectionManager(ILoggerFactory loggerFactory, IApplicationLifetime appLifetime) 
         {
             _logger = loggerFactory.CreateLogger<HttpConnectionManager>();
             _connectionLogger = loggerFactory.CreateLogger<HttpConnectionContext>();
             _nextHeartbeat = new TimerAwaitable(_heartbeatTickRate, _heartbeatTickRate);
-
+            DisconnectTimeout = ConnectionOptionsSetup.DefaultDisconectTimeout;
             // Register these last as the callbacks could run immediately
             appLifetime.ApplicationStarted.Register(() => Start());
             appLifetime.ApplicationStopping.Register(() => CloseConnections());
         }
 
+        public HttpConnectionManager(ILoggerFactory loggerFactory, IApplicationLifetime appLifetime, IOptions<ConnectionOptions> connectionOptions)
```

Comment:
Usually, the one with the most argument does the actual work and the ones with less arguments just delegate to that one.